scapy/supersocket.py

```diff
@@ -400,16 +472,19 @@ class TunTapInterface(SuperSocket):
     desc = "Act as the host's peer of a tun / tap interface"
 
     def __init__(self, iface=None, mode_tun=None, *arg, **karg):
-        self.iface = conf.iface if iface is None else iface
-        self.mode_tun = ("tun" in self.iface) if mode_tun is None else mode_tun
+        # type: (Optional[str], Optional[str], Any, Any) -> None
+        self.iface = iface or conf.iface
+        self.mode_tun = (("tun" in network_name(self.iface))
+                         if mode_tun is None else mode_tun)
         self.closed = True
         self.open()
 
     def open(self):
+        # type: () -> None
         """Open the TUN or TAP device."""
         if not self.closed:
             return
-        self.outs = self.ins = open(
```

Comment:
This was pretty wrong. `self.ins` or `self.outs` are not relevant in this socket.